SourceCode/CRTRenderer.cpp

```diff
@@ -325,3 +331,44 @@ CRTImage CRTRenderer::renderWithAABB() const
     }
     return image;
 }
+
+CRTImage CRTRenderer::renderAccelerated() const
+{
+    const unsigned imageHeight = scene->getSettings().imageSettings.height;
+    const unsigned imageWidth = scene->getSettings().imageSettings.width;
+
+    CRTImage image(imageHeight, std::vector<CRTVector>(imageWidth, scene->getSettings().bgColor));
+
+    CRTBucketQueue queue(scene); // the constructor generates the regions
+
+    int threadCount = std::thread::hardware_concurrency();
+    std::vector<std::thread> threads;
+
+    for (int i = 0; i < threadCount; i++) {
+        threads.push_back(std::thread([this, &queue, &image]() {
+            CRTRegion region;
+            while (true) {
+                if (!queue.getRegion(region)) { // if getRegion returns false, then there's no more buckets
+                    break;
+                }
+                renderRegionAccelerated(region.startX, region.startY, region.width, region.height, image);
+            }
+            }
+        ));
+    }
+
+    for (auto& thread : threads) {
+        thread.join();
+    }
+    return image;
+}
+
+CRTImage CRTRenderer::renderAcceleratedSinglethreaded() const
```

Comment:
It's very good that you continue to maintain the single threaded version of your renderer